Fix rhel-x image

martinpitt opened the pull request:
Fixes #9455

 * [x] image-refresh rhel-x
 * [x] unbreak boot loader install (https://bugzilla.redhat.com/show_bug.cgi?id=1597503)
 * [x] unbreak broken grub config (https://bugzilla.redhat.com/show_bug.cgi?id=1601578)
 * [x] unbreak networking failure (unable to reach machine - https://bugzilla.redhat.com/show_bug.cgi?id=1600823)
 * [x] avoid PackageKit eating all CPU (https://bugzilla.redhat.com/show_bug.cgi?id=1600452)
 * [x] investigate IPA test case failures (https://bugzilla.redhat.com/show_bug.cgi?id=1560452 came back)
 * [x] fix `check-kdump` failure (image now has `crashkernel=auto` by default)
 * [x] fix `check-realms` failures due to sudo (https://bugzilla.redhat.com/show_bug.cgi?id=1577615)
 * [x] fix `check-docker-storage TestDockerStorage.testDevicemapper` regression (PR #9720)
 * [x] `check-networking-firewall TestFirewall.testNetworkingPage` regression (https://bugzilla.redhat.com/show_bug.cgi?id=1607929)
 * [x] ~`check-setroubleshoot` regression~ - deferred to issue #9721

cockpituous commented:
image-refresh in progress on cockpit-tasks-gxm27.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9475-20180623-141919/

martinpitt commented:
I checked interactively and don't see how to fix that:
```
[root@m1 ~]# yum install atomic docker
Last metadata expiration check: 12:15:01 ago on So 24 Jun 2018 02:57:25 EDT.
Error: 
 Problem 1: conflicting requests
  - nothing provides runc needed by atomic-1.20.1-2.el8+5.x86_64
 Problem 2: conflicting requests
  - nothing provides skopeo-containers >= 0.1.25-2 needed by docker-2:1.13.1-54.git4032bd5.el8+7.x86_64
```
I asked on the ML.

martinpitt commented:
This is still blocked as runc and skopeo-containers are not installable. The current  BaseURL is definitively gone - so we just can't build a working rhel-x image right now. Thus we can just as well land this now, and one of the next automatic bot updates will tell us when this works again. WDYT, @larskarlitski ?

martinpitt commented:
runc and skopeo-containers should exist now, retrying build.

cockpituous commented:
image-refresh in progress on cockpit-tasks-dr8pk.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9475-20180702-063535/

martinpitt commented:
Refresh hangs, *sigh*, more needs work.

martinpitt commented:
The boot loader install failure got reported, and cannot be fixed on our side, so back to blocked.

cockpituous commented:
image-refresh in progress on cockpit-tasks-dz9md.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9475-20180702-141818/

cockpituous commented:
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9475-20180702-141818/

martinpitt commented:
Another attempt, the boot loader issue was said to be worked around today.

cockpituous commented:
image-refresh in progress on cockpit-tasks-hjk03.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9475-20180713-061717/

larskarlitski commented:
Shouldn't we run a rhel-x test on this?

martinpitt commented:
@larskarlitski : There's no point as long as the rhel-x image still fails to build. This is still blocked on the installer failure (boot loader).

cockpituous commented:
image-refresh in progress on cockpit-tasks-w2dgv.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9475-20180713-131919/

martinpitt commented:
I can reproduce the current failure. The `virt-customize` command that copies the network config still works; then image-create starts the VM again, and times out waiting to reboot. After starting it, the domain is shown as running:

     3     rhel-x-127.0.0.2-2201          running

but neither ssh nor `virsh console` show/do anything. `virsh reboot` does not work either, only `virsh reset`. Then the console shows `error: invalid magic number. Press any key to continue...`. But pressing "any key" doesn't continue anything, it's just stuck.

I tried booting the resulting VM directly in QEMU, same result

cockpituous commented:
Task aborted.

cockpituous commented:
image-refresh in progress on cockpit-tasks-1f3tj.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9475-20180714-213737/

cockpituous commented:
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9475-20180714-213737/

martinpitt commented:
I found out what breaks the boot, see updated description. I committed a workaround, I can now locally build an image again that works. Re-trying on the bots.

cockpituous commented:
image-refresh in progress on cockpit-tasks-kchxl.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9475-20180717-102222/

cockpituous commented:
Task failed: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9475-20180717-102222/

cockpituous commented:
image-refresh in progress on cockpit-tasks-vsz2l.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9475-20180717-113131/

cockpituous commented:
image-refresh in progress on cockpit-tasks-qg7zm.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9475-20180717-113232/

cockpituous commented:
image-refresh rhel-x done: https://github.com/cockpituous/cockpit/commits/image-refresh-rhel-x-20180717-115151

cockpituous commented:
image-refresh in progress on cockpit-tasks-dqjn3.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9475-20180717-134141/

cockpituous commented:
image-refresh rhel-x done: https://github.com/cockpituous/cockpit/commits/image-refresh-rhel-x-20180717-140303

martinpitt commented:
I pushed a commit to better diagnose the networking failure on bootup, it is hard to reproduce locally.

cockpituous commented:
image-refresh in progress on cockpit-tasks-0ll97.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-9475-20180724-102424/

cockpituous commented:
image-refresh rhel-x done: https://github.com/cockpituous/cockpit/commits/image-refresh-rhel-x-20180724-104545

martinpitt commented:
I *think* I investigated the last regression now. Pushing again and crossing fingers!

martinpitt commented:
@allisonkarlitskaya : I put the more hideous hacks into separate commits, to make them easier to `git revert` later on, when the bugs get fixed. I know this isn't strictly adhering to your "every commit should be self-contained" rule, but at least the image refresh (which is the real trigger of the regression) comes on top, and the hacks would also apply to our current image. But if you prefer, I can squash them together.

Otherwise this was primarily an exercise of "debug lots of random breakage", so not much freedom in how to fix/work around them. Thus the review should be fairly easy. :-)

allisonkarlitskaya commented on test/verify/check-kdump:
```diff
@@ -29,8 +29,8 @@ class TestKdump(MachineCase):
-        if "atomic" in self.machine.image:
-            # on atomic we have a crashkernel option already, but for auto
+        if self.machine.atomic_image or self.machine.image  in ["rhel-x"]:
+            # on atomic and rhel-x we have a crashkernel option already, but for auto
```
can you please assert this, just incase it stops being true at some point?

alternatively, why not just 'sed -i' this?

martinpitt replied:
`sed`ing would be easier indeed, I can change this.

The test already does assert it, in line 77ff. - if. e. g. rhel-x would stop defaulting to this, then the `Service is running` assertion below would be wrong, as it would say "Service has an error" and "No memory reserved".

martinpitt replied:
I added an assertion; might be easier to debug than the failure further below, indeed.

allisonkarlitskaya commented on bots/images/scripts/rhel-x.install:
```diff
@@ -4,3 +4,7 @@ set -e
+
+# HACK: https://bugzilla.redhat.com/show_bug.cgi?id=1600452
+# packagekitd spins at 100% CPU on startup
+systemctl mask packagekit.service
```
This kind of problem seems likely to be fixed relatively soon...  Do we have a way to track upstream bugs so that we don't lose track of hacks like this?

martinpitt replied:
With cockpit being "everyone's QA", I file a lot of them :-) For others I subscribe to these, so that I know when they get fixed, and can also help to verify them. Most of these hacks get tracked by naughty overrides that get cleaned up automatically, but this doesn't work here.

martinpitt commented:
Force-pushed again. I added the check-kdump assertion (as above), added a `Closes #` (whoops!), and I dropped the `rhel-domainname.service` hack again -- last night's nightly fixed that, so it's not necessary any more \o/.
